src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java

```diff
@@ -250,52 +255,55 @@ String computeCrc32c(String input) {
         return outputs;
     }
 
-    static boolean shouldRetry(Throwable throwable) {
+    // For batch operations, we want to also include the AbortedException as retryable
+    static boolean shouldRetry(Throwable throwable, boolean isBatch) {
         if (throwable == null) {
             return false; // Did not find a retry-able exception
         }
         if (throwable instanceof DeadlineExceededException ||
             throwable instanceof UnavailableException ||
-            throwable instanceof AbortedException ||
             throwable instanceof InternalException ||
             throwable instanceof StatusRuntimeException) {
 
+            return true;
+        } else if (isBatch && throwable instanceof AbortedException) {
```

Comment:
This clause can be added to the above `if` statement since they both result in returning `true`